core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -0,0 +1,405 @@
+
+  @Override
+  public Set<Class<? extends Server>> getDependencies() {
+    return DEPS;
+  }
+
+  @Override
+  public void processJournalEntry(JournalEntry entry) throws IOException {
+    throw new IOException(ExceptionMessage.UNEXPECTED_JOURNAL_ENTRY.getMessage(entry));
+  }
+
+  @Override
+  public void resetState() {}
+
+  @Override
+  public Iterator<JournalEntry> getJournalEntryIterator() {
+    return null;
+  }
+
+  @Override
+  public void start(Boolean isPrimary) throws IOException {
+    super.start(isPrimary);
+    mWorkerConfigChecker.reset();
+    mStartTimeMs = System.currentTimeMillis();
+    if (isPrimary) {
+      mSafeModeManager.notifyPrimaryMasterStarted();
+
+      //  The service that detects lost standby master nodes
+      getExecutorService().submit(new HeartbeatThread(
+          HeartbeatContext.MASTER_LOST_MASTER_DETECTION,
+          new LostMasterDetectionHeartbeatExecutor(),
+          (int) Configuration.getMs(PropertyKey.MASTER_HEARTBEAT_INTERVAL_MS)));
+    } else {
+      // Standby master should setup MetaMasterSync to communicate with the leader master
+      setMasterId();
+      MetaMasterSync metaMasterSync =
+          new MetaMasterSync(mMasterId, mMasterHostname, mMetaMasterClient);
+      getExecutorService().submit(new HeartbeatThread(HeartbeatContext.META_MASTER_SYNC,
+          metaMasterSync, (int) Configuration.getMs(PropertyKey.MASTER_HEARTBEAT_INTERVAL_MS)));
+      LOG.info("Standby master with id {} starts sending heartbeat to leader master.", mMasterId);
+    }
+  }
+
+  @Override
+  public void stop() throws IOException {
+    super.stop();
+  }
+
+  @Override
+  public List<ConfigProperty> getConfiguration() {
+    List<ConfigProperty> configInfoList = new ArrayList<>();
+    String alluxioConfPrefix = "alluxio";
+    for (Map.Entry<String, String> entry : Configuration.toMap().entrySet()) {
+      String key = entry.getKey();
+      if (key.startsWith(alluxioConfPrefix)) {
+        PropertyKey propertyKey = PropertyKey.fromString(key);
+        String source = Configuration.getFormattedSource(propertyKey);
+        configInfoList.add(new ConfigProperty()
+            .setName(key).setValue(entry.getValue()).setSource(source));
+      }
+    }
+    return configInfoList;
+  }
+
+  @Override
+  public long getMasterId(String hostname) {
+    MasterInfo existingMaster = mMasters.getFirstByField(HOSTNAME_INDEX, hostname);
+    if (existingMaster != null) {
+      // This master hostname is already mapped to a master id.
+      long oldMasterId = existingMaster.getId();
+      LOG.warn("The master {} already exists as id {}.", hostname, oldMasterId);
+      return oldMasterId;
+    }
+
+    MasterInfo lostMaster = mLostMasters.getFirstByField(HOSTNAME_INDEX, hostname);
+    if (lostMaster != null) {
+      // This is one of the lost masters
+      mMasterConfigChecker.lostNodeFound(lostMaster.getId());
+      synchronized (lostMaster) {
+        final long lostMasterId = lostMaster.getId();
```

Comment:
is the final needed?

Reply:
Removed, thanks!
It is strange why the previous people using ```final long lostWorkerId``` in the DefaultBlockMaster.getWorkerId().

Reply:
I think that code is old. In java 8+ there's really never a reason to declare a local variable `final`